Sources/SymbolKit/SymbolGraph/Symbol/Symbol.swift

```diff
@@ -58,7 +58,12 @@ extension SymbolGraph {
         /// the same module as the symbol or not.
         ///
         /// An inherited documentation comment is from the same module when the symbol that the documentation is inherited from is in the same module as this symbol.
+        @available(*, deprecated, message: "Use 'docComment?.moduleName' instead to compare the documentation's source module name with another module name.")
```

Comment:
I'm having second thoughts about deprecating this. 

On the one hand we want to inform developers that there is a more accurate way to determine the source module of a documentation comment.

One the other hand, developers may need to support symbol graph files that predate the new source module information for some time, leading to unavoidable deprecation warnings. 

At some point we want to remove this property but that may not be possible until the new data has been available for some time.

Reply:
Actually, if I move the fallback logic into a new function in SymbolKit, then clients can avoid using the deprecated property in favor of the function.